cloud/store/bootstrap.go

```diff
@@ -0,0 +1,87 @@
+package store
+
+import (
+	"context"
+	"log"
+	"math/rand"
+	"time"
+
+	cmd "github.com/weaviate/weaviate/cloud/proto/cluster"
+)
+
+type joiner interface {
+	Join(ctx context.Context, leaderAddr string, req *cmd.JoinPeerRequest) (*cmd.JoinPeerResponse, error)
+	Notify(ctx context.Context, leaderAddr string, req *cmd.NotifyPeerRequest) (*cmd.NotifyPeerResponse, error)
+}
+
+// Bootstrapper is used to bootstrap this node by attempting to join it to a RAFT cluster.
+type Bootstrapper struct {
```

Comment:
I understand what you mean by Bootstrap in this context, but we should be careful or clearly define the terms we use.

In the raft context, bootstrap means bootstrapping a cluster. Is this bootstrapper used to bootstrap a whole cluster or only to join a given existing one ? Is it doing both ? Maybe we can have a clearer name either on the struct itself, or on the functions we use on this struct to clearly state it's purpose.

Reply:
The bootstrapper focuses on bootsrapping this node, and joining it to cluster so it can be functional. The result of the bootstrapping is a functional node which align well with the definition